app/schema/preference.ts

```diff
@@ -0,0 +1,11 @@
+import { z } from "zod";
+
+export default z.object({
+  body: z.object({
+    preference: z.object({
+      darkmode: z.boolean({
+        required_error: "Boolean required",
+      }),
+    }),
+  }),
+});
```

Comment:
Also double check that you can't add any additional fields.

Reply:
Since my route was same as method, I had to name it PreferenceGetGet.

Also checked that you can't add any more fields due to update function in PreferenceService.ts

``` ts
async update(uid: string, data: JSON): Promise<JSON> {
    try {
      //@ts-ignore
      const pref = data.preference;
      await prisma.preference.update({
        where: {
          uid,
        },
        data: pref,
      });
      return data;
    } catch (e) {
      throw new UnprocessableEntityException();
    }
  },
```

This throws the UnprocessableEntityException since the data does not match to the one in the database

Reply:
👀  wait, so this was input checking a get request? 

And awesome, so we don't have to worry about extra fields being slid in.